glib/src/gstring.rs

```diff
@@ -511,6 +534,21 @@ mod tests {
         assert_eq!(s.as_str(), "foo");
     }
 
+    #[test]
+    fn test_from_glib_container() {
+        unsafe {
+            let a: GString =
+                FromGlibContainer::from_glib_container_num("hello_world".as_ptr() as *const i8, 5);
+            assert_eq!("hello", a.as_str());
+            let b: GString =
+                FromGlibContainer::from_glib_full_num("hello_world".as_ptr() as *const i8, 5);
```

Comment:
```suggestion
                FromGlibContainer::from_glib_full_num(glib_sys::g_strdup("hello_world".as_ptr() as *const i8), 5);
```

Or similar. The problem is that you pass a static (and definitely not heap allocated with the GLib allocator) string in there but say that `GString` can take ownership of it. That doesn't work :)

Same for the container variant as they're really the same here.

Reply:
I should have looked at the other tests first..